File: CourseWorkOnAndroidAndSpring-main/project/server/src/main/java/ru/ryabov/studentperformance/dto/journal/LessonRecordDto.java
package ru.ryabov.studentperformance.dto.journal;

import java.time.LocalDate;
import java.util.List;

/** Одно занятие в журнале: дата, группа, предмет, тип, список студентов (присутствие + баллы). Id — для ссылки «Редактировать». */
public class LessonRecordDto {
    private final LocalDate date;
    private final String groupName;
    private final String subjectName;
    private final String gradeTypeName;
    private final List<StudentLessonRowDto> students;
    private final Long groupId;
    private final Long subjectId;
    private final Long gradeTypeId;

    public LessonRecordDto(LocalDate date, String groupName, String subjectName, String gradeTypeName, List<StudentLessonRowDto> students) {
        this(date, groupName, subjectName, gradeTypeName, students, null, null, null);
    }

    public LessonRecordDto(LocalDate date, String groupName, String subjectName, String gradeTypeName, List<StudentLessonRowDto> students, Long groupId, Long subjectId, Long gradeTypeId) {
        this.date = date;
        this.groupName = groupName;
        this.subjectName = subjectName;
        this.gradeTypeName = gradeTypeName;
        this.students = students != null ? students : List.of();
        this.groupId = groupId;
        this.subjectId = subjectId;
        this.gradeTypeId = gradeTypeId;
    }

    public LocalDate getDate() { return date; }
    public String getGroupName() { return groupName; }
    public String getSubjectName() { return subjectName; }
    public String getGradeTypeName() { return gradeTypeName; }
    public List<StudentLessonRowDto> getStudents() { return students; }
    public Long getGroupId() { return groupId; }
    public Long getSubjectId() { return subjectId; }
    public Long getGradeTypeId() { return gradeTypeId; }
}

File: CourseWorkOnAndroidAndSpring-main/project/mobile/app/src/main/java/ru/zagrebin/studentperformance/data/remote/dto/LessonRecordDto.kt
package ru.ryabov.studentperformance.data.remote.dto

import com.google.gson.annotations.SerializedName

/** Одно занятие в журнале: дата, группа, предмет, тип, список студентов. */
data class LessonRecordDto(
    @SerializedName("date") val date: String? = null,
    @SerializedName("groupName") val groupName: String? = null,
    @SerializedName("subjectName") val subjectName: String? = null,
    @SerializedName("gradeTypeName") val gradeTypeName: String? = null,
    @SerializedName("students") val students: List<StudentLessonRowDto>? = null,
    @SerializedName("groupId") val groupId: Long? = null,
    @SerializedName("subjectId") val subjectId: Long? = null,
    @SerializedName("gradeTypeId") val gradeTypeId: Long? = null
)

/** Строка по студенту в занятии: имя, присутствие, баллы. */
data class StudentLessonRowDto(
    @SerializedName("studentName") val studentName: String? = null,
    @SerializedName("present") val present: Boolean = false,
    @SerializedName("points") val points: java.math.BigDecimal? = null
)

/** Ответ API journal/lesson-records: страница занятий. */
data class LessonRecordsPageDto(
    @SerializedName("content") val content: List<LessonRecordDto>? = null,
    @SerializedName("totalElements") val totalElements: Long = 0,
    @SerializedName("totalPages") val totalPages: Int = 0,
    @SerializedName("number") val number: Int = 0,
    @SerializedName("size") val size: Int = 0
)
